File: LibreOJ/6/6.cpp
#include <bits/stdc++.h>

#include "interaction.h"

using namespace std;

int n;
vector<int> ans;

int main() {
    n = get_num();
    for (int i = 0; i < n; i++) {
        int l = 0, r = 1000000;

        while (l <= r) {
            int mid = (l + r) >> 1, feedback = guess(i, mid);

            if (!feedback) {
                ans.emplace_back(mid);
                break;
            } else {
                if (feedback > 0) {
                    r = mid - 1;
                } else {
                    l = mid + 1;
                }
            }
        }
    }
    submit(ans);
    return 0;
}

File: LibreOJ/6/6.go
package main

import "fmt"

func main() {
    var n int
    var ans [105]int

    fmt.Println("get_num")
    fmt.Scan(&n)

    for i := 0; i < n; i++ {
        l, r := 1, 1000000

        for l <= r {
            mid := (l + r) >> 1
            fmt.Println("guess", i, mid)

            var res int
            fmt.Scan(&res)

            if res == 0 {
                ans[i] = mid
                break
            } else if res > 0 {
                r = mid - 1
            } else { // res < 0
                l = mid + 1
            }
        }
    }

    fmt.Print("submit ")
    for i := 0; i < n; i++ {
        fmt.Printf("%d ", ans[i])
    }
    fmt.Printf("\n")
}
